core/src/main/scala/org/apache/spark/util/UninterruptibleThread.scala

```diff
@@ -101,7 +101,11 @@ private[spark] class UninterruptibleThread(
      *         is true) and no concurrent [[interrupt()]] call, otherwise false
      */
     def isInterruptible: Boolean = synchronized {
-      shouldInterruptThread = uninterruptible
+      // SPARK-53394: We should not interrupt the thread when it is already interrupted.
+      // Otherwise, the state of `shouldInterruptThread` becomes inconsistent between
```

Comment:
nit: this is code comment, not a doc comment, using Markdown formatting reduces readability IMO.